localstack-core/localstack/deprecations.py

```diff
@@ -275,6 +275,17 @@ def is_affected(self) -> bool:
         "This option was confusingly named. Please use DNS_NAME_PATTERNS_TO_RESOLVE_UPSTREAM "
         "instead.",
     ),
+    EnvVarDeprecation(
```

Comment:
Probably better left for a seperate discussion but should we not rather have a `EnvVarUnsupported` function since deprecation sounds like we're just no longer supporting it.

Reply:
Thanks for thinking about how we can improve customer communication here. +1, better to discuss this in #sig-localstack (or prj-v4 if related to the upcoming release).